core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -134,8 +134,13 @@ final class JdbcUtil {
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
+          + " ( "
           + TABLE_NAMESPACE
-          + " LIKE ? LIMIT 1";
+          + " = ? OR "
```

Comment:
thank you for separate out for both equality check and prefix check with escape.